pkg/deploy/registry/deploylog/rest.go

```diff
@@ -180,6 +186,31 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 	}, nil
 }
 
+// waitForExistingDeployment will use the timeout to wait for a deployment to appear.
+func (r *REST) waitForExistingDeployment(namespace, name string) (*kapi.ReplicationController, error) {
+	var (
+		target *kapi.ReplicationController
+		err    error
+	)
+
+	condition := func() (bool, error) {
```

Comment:
Actually, on a second thought I am not sure how we should go about this. `oc logs` without specifying a version lets the server decide for which version should it return logs... The server decide which container endpoint to hit by taking its best guess. If it starts returning logs for the 3rd deployment and a moment later something increments latestVersion, then you want to rerun `oc logs` because the 3rd deployer just exited, no?